### ISSUE TITLE
failed to reimplement

### ISSUE THREAD

**Issue by YiZhuo-Xu** [Jam1ezhang/RankCLIP#2]
First of all, thank you for your work. Although there is no code, I tried to reproduce some of the results based on your paper and encountered some issues. I found that the loss function based on the ranking model seems not to converge during actual training. One possibility that occurred to me is that the product of rank probabilities is too small, leading to numerical instability. I would like to know if the implementation provided in the paper is still an incomplete version. I am looking forward to your reply very much.
